typegate/tests/prisma/full_prisma_mapping_test.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright Metatype OÜ under the Elastic License 2.0 (ELv2). See LICENSE.md for usage.
+
+import { gql, recreateMigrations, test } from "../utils.ts";
+
+test("prisma full mapping", async (t) => {
+  const e = await t.pythonFile("prisma/full_prisma_mapping.py");
+
+  await t.should("drop schema and recreate", async () => {
+    await gql`
+      mutation a {
+        dropSchema
+      }
+    `
+      .expectData({
+        dropSchema: 0,
+      })
+      .on(e);
+    await recreateMigrations(e);
+  });
+
+  // adding test datas
+  await t.should("insert a record with nested object", async () => {
+    await gql`
+        mutation q {
+          createOneUser (
+            data: {
+              id: 1
+              name: "Jack"
+              age: 20
+              coinflips: [false, true, true]
+              city: "Anyville"
+              posts: {
+                createMany: {
+                  data: [
+                    { id: 10001, title: "Book 1", views: 9, likes: 7, published: true },
+                    { id: 10002, title: "Book 2", views: 8, likes: 3, published: false },
+                    { id: 10003, title: "Book 3", views: 5, likes: 20, published: true },
+                    { id: 10004, title: "Book 4", views: 5, likes: 14, published: true },
+                  ]
+                }
+              }
+              extended_profile: {
+                create: { id: 10111, bio: "Some bio 1" }
+              }
+            }
+          ) 
+        {
+          id
+        }
+      }
+    `.expectData({
+      createOneUser: { id: 1 },
+    })
+      .on(e);
+  });
+
+  await t.should("do a count with findManyPost", async () => {
+    await gql`
+        query {
+          findManyUsers {
+            _count { posts extended_profile }
+          }
+        }
+    `.expectBody((body: any) => {
+      console.log("body findManyPost :::", body);
+    })
+      .on(e);
+  });
+
+  await t.should("do a count with findUniquePost", async () => {
+    await gql`
+        query {
+          findUniquePost {
+            _count { posts extended_profile }
+          }
+        }
+    `.expectBody((body: any) => {
+      console.log("body findManyPost :::", body);
+    })
+      .on(e);
+  });
+
+  await t.should("paginate correctly with findManyPosts", async () => {
+    await gql`
+        query {
+          findManyPosts(skip: 1, take: 2) {
+            id 
+            title
+          }
+        }
+    `.expectData({
+      findManyPosts: [
+        { id: 10002, title: "Book 2" },
+        { id: 10003, title: "Book 3" },
+      ],
+    })
+      .on(e);
+  });
+
+  await t.should("orderBy likes and views", async () => {
```

Comment:
I would include some entries with the same `likes` but different `views` to see if it is ordered correctly by `views`.